ev3dev/control/GyroBalancer.py

```diff
@@ -100,383 +105,379 @@ def __init__(self,
-                            steering, frictionOffset, voltageCompensation)
-                    SetDuty(motorDutyCycleLeft, motorDutyCycle - steering,
-                            frictionOffset, voltageCompensation)
-
-                    ###########################################################
-                    #  Update angle estimate and Gyro Offset Estimate
-                    ###########################################################
-
-                    gyroEstimatedAngle = gyroEstimatedAngle + gyroRate *\
-                        loopTimeSec
-                    gyroOffset = (1 - gyroDriftCompensationRate) *\
-                        gyroOffset + gyroDriftCompensationRate * gyroRateRaw
-
-                    ###########################################################
-                    #  Update Accumulated Motor Error
-                    ###########################################################
-
-                    motorAngleErrorAccumulated = motorAngleErrorAccumulated +\
-                        motorAngleError*loopTimeSec
-
-                    ###########################################################
-                    #  Read the touch sensor (the kill switch)
-                    ###########################################################
-
-                    touchSensorPressed = FastRead(self.touchSensorValueRaw)
-
-                    ###########################################################
-                    #  Busy wait for the loop to complete
-                    ###########################################################
-
-                    while(time.time() - tLoopStart < loopTimeSec):
-                        time.sleep(0.0001)
-
-                ###############################################################
-                #
-                # Closing down & Cleaning up
-                #
-                ###############################################################
-
-                # Loop end time, for stats
-                tProgramEnd = time.time()
-
-                # Turn off the motors
-                FastWrite(motorDutyCycleLeft, 0)
-                FastWrite(motorDutyCycleRight, 0)
-
-                # Wait for the Touch Sensor to be released
-                while self.touchSensor.is_pressed:
-                    time.sleep(0.01)
-
-                # Calculate loop time
-                tLoop = (tProgramEnd - tProgramStart)/loopCount
-                log.info("Loop time:" + str(tLoop*1000) + "ms")
-
-                # Print a stop message
-                log.info("-----------------------------------")
-                log.info("STOP")
-                log.info("-----------------------------------")
-
-        # Exit cleanly so that all motors are stopped
-        except (KeyboardInterrupt, Exception) as e:
-            log.exception(e)
-            shutdown()
+        self.motor_left = LargeMotor(left_motor_port)
+        self.motor_right = LargeMotor(right_motor_port)
```

Comment:
Allow the use to pass in the motor class to use but have it default to LargeMotor? I can’t think of any balancers that use medium motors but hey maybe we could build one :)